tutorials/containers/using-helm-with-canonical-kubernetes/using-helm-with-canonical-kubernetes.md

```diff
@@ -0,0 +1,122 @@
+---
+id: using-helm-with-canonical-kubernetes
+summary: Learn how to deploy Helm on Charmed Kubernetes
+categories: containers
+tags: juju, kubernetes, beginner, container, helm, charts
+difficulty: 3
+status: draft
+author: Natalia Litvinova <natalia.litvinova@canonical.com>
+feedback_url: https://github.com/canonical-websites/tutorials.ubuntu.com/issues
+published: 2019-12-03
+
+---
+# Using Helm with Canonical Kubernetes
+
+## Overview
+Duration: 2:00
+
+Helm is a tool that streamlines installing and managing Kubernetes applications. Think of it like apt/yum/homebrew for Kubernetes. It allows describing the application structure through convenient helm-charts and managing it with simple commands.
+
+### What you'll learn
+
+- How to deploy Helm 
+- How to deploy a chart
+
+### You will only need
+
+- a Charmed Kubernetes cluster
+
+positive
+: If you do not have a Charmed Kubernetes cluster, you can refer to the following [tutorial](https://tutorials.ubuntu.com/tutorial/get-started-charmed-kubernetes) to spin up one in minutes. Charmed Kubernetes is a production-grade Kubernetes offering from Canonical which is fully compliant with the upstream project. Get Charmed Kubernetes right away and benefit from simplified deployments and operations provided by Juju charms.
+
+Survey
+: How will you use this tutorial?
+ - Only read through it
+ - Read it and complete the exercises
+: What is your current level of experience?
+ - Novice
+ - Intermediate
+ - Proficient
+
+## Prepare your cluster
+Duration: 2:00
+
+Make sure that your Kubernetes cluster is running and kubectl config is in ~/.kube/config.
+
+`juju scp kubernetes-master/0:config ~/.kube/config`
+
+Also make sure you have opened the Helm port 44134 on all your worker nodes:
+
+```
+juju run --unit kubernetes-worker/0 "open-port 44134"
```

Comment:
Very recently, `juju run` was replaced by `juju exec`. At the moment, they both work. You can ask Juju to exec a command across every unit of an application in a single line:

```
juju exec --app kubernetes-worker "open-port 44134"
```